selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/requestfactory/TestAdminHttpJsonRequestFactory.java

```diff
@@ -10,21 +10,32 @@
  */
 package org.eclipse.che.selenium.core.requestfactory;
 
-import com.google.inject.Inject;
-import com.google.inject.Provider;
-import org.eclipse.che.selenium.core.user.AdminTestUser;
+import com.google.inject.name.Named;
+import javax.inject.Inject;
+import org.eclipse.che.selenium.core.client.TestAuthServiceClient;
 
 /** @author Dmytro Nochevnov */
 public class TestAdminHttpJsonRequestFactory extends TestHttpJsonRequestFactory {
 
-  private Provider<AdminTestUser> adminProvider;
+  private final String adminEmail;
+  private final String adminPassword;
+  private final TestAuthServiceClient authServiceClient;
 
   @Inject
-  public TestAdminHttpJsonRequestFactory(Provider<AdminTestUser> adminProvider) {
-    this.adminProvider = adminProvider;
+  public TestAdminHttpJsonRequestFactory(
+      TestAuthServiceClient authServiceClient,
+      @Named("che.admin_user.email") String adminEmail,
```

Comment:
It's Che-specific settings which should be re-written in tests for Codenvy onprem/saas. 
I propose to create **TestAdminEmailProvider** and **TestAdminPasswordProvider** interfaces and next implementations for them:
- **CheTestAdminEmailProvider** and **CheTestAdminPasswordProvider** implementations in selenium tests for Che (where che.admin_user.email and che.admin_user.password" are using);
- **CodenvyTestAdminEmailProvider** and **CodenvyTestAdminPasswordProvider** implementations for Codenvy (where codenvy.admin.name, codenvy.admin.initial_password, and codenvy.admin.password similarly to the class https://github.com/codenvy/codenvy/blob/master/selenium/codenvy-selenium-test/src/main/java/com/codenvy/selenium/core/user/OnpremAdminTestUser.java)